contentpool_client.module

```diff
@@ -94,3 +95,40 @@ function contentpool_client_theme($existing, $type, $theme, $path) {
     ],
   ];
 }
+
+/**
+ * Implements hook_page_attachments_alter().
+ *
+ * If host from canonical url replicated from contentpool site does not match current host, add canonical url to metadata.
+ */
+function contentpool_client_page_attachments_alter(array &$attachments) {
+  $node = \Drupal::routeMatch()->getParameter('node');
+  if ($node instanceof NodeInterface) {
+    if ($node->hasField('field_canonical_url') && !$node->get('field_canonical_url')->isEmpty()) {
+      $canonical_url = $node->field_canonical_url[0]->uri;
+      $canonical_host = parse_url($canonical_url, PHP_URL_HOST);
```

Comment:
hm, while this should work in 90% in the cases, it will fail when two satellites are deployed on the same domain in different folders. better just generate the base url (Url::fromUri('base:/')) and check whether the URL starts with it. If not, then the URL is from another site.